lib/icingadb/icingadbchecktask.cpp

```diff
@@ -361,33 +382,11 @@ void IcingadbCheckTask::ScriptFunc(const Checkable::Ptr& checkable, const CheckR
 	}
 
 	{
-		static boost::regex wellNamedUnits (":(bytes|seconds)$");
-		ObjectLock lock (goMetricsByCumulativity);
-
-		for (auto& kv : goMetricsByCumulativity) {
-			bool cumulative = kv.first == "cumulative";
-			Dictionary::Ptr goMetricsPerCumulativity = kv.second;
-			ObjectLock lock (goMetricsPerCumulativity);
-
-			for (auto& kv : goMetricsPerCumulativity) {
-				std::string unit;
-				boost::smatch what;
-
-				if (boost::regex_search(kv.first.GetData(), what, wellNamedUnits)) {
-					unit = what[1];
-				}
-
-				bool counter = cumulative && unit.empty();
-				auto label ("go" + kv.first);
+		Array::Ptr values = PluginUtility::SplitPerfdata(perfdataFromRedis);
+		ObjectLock lock (values);
 
-				for (auto& c : label) {
-					if (!('a' <= c && c <= 'z' || 'A' <= c && c <= 'Z' || '0' <= c && c <= '9')) {
-						c = '_';
-					}
-				}
-
-				perfdata->Add(new PerfdataValue(std::move(label), kv.second, counter, std::move(unit)));
-			}
+		for (const auto& v : values) {
```

Comment:
At your opinion: if you can s/const auto/auto/, s/const auto/auto/. After all your intention behind auto is to make things shorter, not longer, isn’t it?

Reply:
Oh, not for actual constants like const auto x (10); of course!